bentoml/deployment/sagemaker/__init__.py

```diff
@@ -223,6 +226,55 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 }
 
 
+def _parse_aws_client_exception_or_raise(e):
+    """parse botocore.exceptions.ClientError into Bento StatusProto
+
+    We handle two most common errors when deploying to Sagemaker.
+        1. Authenication issue/invalid access(InvalidSignatureException)
+        2. resources not found (ValidationException)
+    It will return correlated StatusProto(NOT_FOUND, UNAUTHENTICATED)
+
+    Args:
+        e: ClientError from botocore.exceptions
+    Returns:
+        StatusProto
+    """
+    error_response = e.response.get('Error', {})
+    error_code = error_response.get('Code')
+    if error_code == 'ValidationException':
+        logger.info(str(error_response))
+        return Status.NOT_FOUND(error_response.get('Message', 'Unknown'))
+    elif error_code == 'InvalidSignatureException':
+        logger.info(str(error_response))
+        return Status.UNAUTHENTICATED(error_response.get('Message', 'Unknown'))
```

Comment:
`else raise e` is still necessary here, to make sure when it gets an AWS client error that we don't handle properly